projects/packages/mcp/src/AbilitiesRegistry/Executors/Post/PostGetExecutor.php

```diff
@@ -0,0 +1,254 @@
+<?php // phpcs:ignore
+
+namespace Automattic\Jetpack\AbilitiesRegistry\Executors\Post;
+
+use Automattic\Jetpack\AbilitiesRegistry\Helpers\PostQueryHelper;
+use Automattic\Jetpack\AbilitiesRegistry\Interfaces\ExecutorInterface;
+use Exception;
+
+/**
+ * Post Get Executor
+ *
+ * Contains execution logic for single post retrieval operations with site switching support.
+ * Only loaded when the ability is actually executed.
+ */
+class PostGetExecutor implements ExecutorInterface {
+
+	/**
+	 * Execute the post get ability.
+	 *
+	 * @param array $input The input parameters.
+	 *
+	 * @return array The post data or error response.
+	 */
+	public function execute( array $input = array() ): array {
+		$original_blog_id = get_current_blog_id();
+		$switched         = false;
+
+		try {
```

Comment:
I think we generally try to avoid using try/catch, so we can more easily see what isn’t working correctly.

Reply:
I'm going to leave them in place for now.